### PR TITLE
WIP feat(ngm): added npm-submodule build process

### PR DESCRIPTION
**Pull request by valorkin**

### CODE REVIEW

File: src/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    "suppressExcessPropertyErrors": true,
+    "suppressImplicitAnyIndexErrors": true,
+    "outDir": "../dist",
+    "lib": ["dom", "es6"],
+    "types": [
+      "jasmine",
+      "webpack"
+    ],
+    "typeRoots": [
+      "../node_modules/@types"
```
Comment (amcdnl) [inline src/tsconfig.json]: Why does it need to do this?

File: src/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    "types.d.ts",
+    "index.ts"
+  ],
+  "exclude": [
+    "node_modules"
+  ],
+  "compileOnSave": false,
+  "buildOnSave": false,
+  "awesomeTypescriptLoaderOptions": {
+    "forkChecker": true,
```
Comment (amcdnl) [inline src/tsconfig.json]: Does fork checker work for you? It never seems to actually update for me.

File: src/swui.module.ts [hunk 3]
```diff
@@ -7,7 +7,7 @@ import { DblClickCopyDirective } from './directives';
 import { PipesModule } from './pipes';
 
 import {
-  CalendarModule, CodemirrorModule, CodeHighlightModule,
+  CalendarModule, /*CodemirrorModule,*/ CodeHighlightModule,
```
Comment (amcdnl) [inline src/swui.module.ts]: :(

File: src/components/date-time/date-time.component.ts [hunk 4]
```diff
@@ -22,7 +22,125 @@ const DATE_TIME_VALUE_ACCESSOR = {
 @Component({
   selector: 'swui-date-time',
   providers: [DATE_TIME_VALUE_ACCESSOR],
-  template
+  template: `<div class="swui-date-time">
```
Comment (amcdnl) [inline src/components/date-time/date-time.component.ts]: Why did you bring this inline?

File: src/tsconfig.json (hunk 1 above)
Comment (valorkin) [inline src/tsconfig.json]: In order to replace tsd with @types was added typeRoots
https://github.com/Microsoft/TypeScript/issues/9184
to simplify declarations resolution

File: src/tsconfig.json (hunk 2 above)
Comment (valorkin) [inline src/tsconfig.json]: one day it will work :)

File: src/swui.module.ts (hunk 3 above)
Comment (valorkin) [inline src/swui.module.ts]: We can help code mirror to became AoT compatible too ;)
and than uncomment

File: src/components/date-time/date-time.component.ts (hunk 4 above)
Comment (valorkin) [inline src/components/date-time/date-time.component.ts]: @angular/compiler-cli doesn't understand html imports and can not analyze them staticaly

Comment (amcdnl) [inline src/components/date-time/date-time.component.ts]: ugh ... I suppose thats why they have that custom dealie in the component defs :(

**Comment by amcdnl** [comment]
closing for now.

**Pull request closed without merging**